integrations/Iterable/browser.js

```diff
@@ -0,0 +1,211 @@
+        this.topOffset,
+        this.bottomOffset,
+        this.handleLinks,
+        this.closeButtonColor,
+        this.closeButtonSize,
+        this.closeButtonColorTopOffset,
+        this.closeButtonColorSideOffset,
+        this.iconPath,
+        this.isRequiredToDismissMessage,
+        this.closeButtonPosition,
+    */
+   // Reference : https://github.com/iterable/iterable-web-sdk
+    let getInAppMessagesPayload =  {
+        count: 20,
+        animationDuration: Number(this.animationDuration) || 400,
+        displayInterval: Number(this.displayInterval) || 30000,
+        onOpenScreenReaderMessage: this.onOpenScreenReaderMessage || undefined,
+        onOpenNodeToTakeFocus: this.onOpenNodeToTakeFocus || undefined,
+        packageName: this.packageName || undefined,
+        rightOffset: this.rightOffset || undefined,
+        topOffset: this.topOffset || undefined,
+        bottomOffset: this.bottomOffset || undefined,
+        handleLinks: this.handleLinks || undefined,
+        closeButton: {
+            color: this.closeButtonColor || 'red',
+            size: this.closeButtonSize || '16px',
+            topOffset: this.closeButtonColorTopOffset || '4%',
+            sideOffset: this.closeButtonColorSideOffset || '4%',
+            iconPath: this.iconPath || undefined,
+            isRequiredToDismissMessage: this.isRequiredToDismissMessage || undefined,
+            position: this.closeButtonPosition || 'top-right'
+          }
+        }
+    getInAppMessagesPayload = removeUndefinedAndNullValues(getInAppMessagesPayload);
+
+    const { request } = window['@iterable/web-sdk'].getInAppMessages(
+        getInAppMessagesPayload,
+        { display: 'immediate' }
+        );
+    // fetchAppEvents is a class function now available throughout
+    // we will trigger getInAppMessages when event name matches from the ui mapping in config.
+    this.fetchAppEvents = request;
+  }
+
+  track(rudderElement) {
+    logger.debug("===In track Iterable===");
+    
+    const { message } = rudderElement;
+    const { event } = message;
+    const eventPayload = removeUndefinedAndNullValues(message.properties);
+    const userEmail = get(message, "context.traits.email");
+    const userId = get(message, "userId");
+    
+    if (!event) {
+        logger.error("Event name not present");
+        return;
+    }
+    let isCustom = true;
+    if (isNotEmpty(this.getInAppEventMapping)) {
+        const mappedEvents = this.getInAppEventMapping;
+        mappedEvents.forEach((e) => {
+            if(e.eventName === event) {
+                // send a track call for getinappMessages if option enabled in config
+                if (this.sendTrackForInapp) {
+                    window['@iterable/web-sdk'].track({ email: userEmail, userId, eventName: "Track getInAppMessages", dataFields: eventPayload })
+                    .then(logger.debug("Track a getinappMessages event."));
+                }
+                this.fetchAppEvents();
```

Comment:
can we call `fetchAppEvents` before the tracking?

Reply:
Sure.

Reply:
done